### PR TITLE
Added check for low mach preconditioner sensor [SCFD-2210]

### PR DESCRIPTION
**Pull request by andrzej-krupka**

### CODE REVIEW

**Review by benflexcompute — DISMISSED** [review]

File: flow360/component/simulation/validation/validation_simulation_params.py [hunk 1]
```diff
@@ -36,3 +37,22 @@ def _check_consistency_wall_function_and_surface_output(v):
                 )
 
     return v
+
+
+def _check_low_mach_preconditioner_output(v):
+    models = v.models
```
[the remark this answers is not in the document]
Comment (andrzej-krupka) [reply flow360/component/simulation/validation/validation_simulation_params.py:43]: Fixed in [8bb0f89](https://github.com/flexcompute/Flow360/pull/443/commits/8bb0f894657f28936d45ce054c67a9e88de1a439)

File: flow360/component/simulation/validation/validation_simulation_params.py [hunk 2]
```diff
@@ -36,3 +37,22 @@ def _check_consistency_wall_function_and_surface_output(v):
...
+    for model in models:
+        if isinstance(model, Fluid) and model.navier_stokes_solver:
+            preconditioner = model.navier_stokes_solver.low_mach_preconditioner
+            if preconditioner:
+                continue
+
+            outputs = v.outputs
+            for output in outputs:
```
[the remark this answers is not in the document]
Comment (andrzej-krupka) [reply flow360/component/simulation/validation/validation_simulation_params.py]: Fixed in [8bb0f89](https://github.com/flexcompute/Flow360/pull/443/commits/8bb0f894657f28936d45ce054c67a9e88de1a439)

File: flow360/component/simulation/validation/validation_simulation_params.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (benflexcompute) [reply flow360/component/simulation/validation/validation_simulation_params.py:43]: `if not models:` should be equivalent to `has_low_mach_preconditioner = False`. Instead current validation just returns and skip the checks.

It is very unlikely that we will ever encounter `models=None` but the logic should be correct.

File: tests/simulation/params/test_validators_params.py [hunk 3]
```diff
@@ -87,7 +87,7 @@ def test_low_mach_preconditioner_validator(
     assert params
 
     message = (
-        "Low-Mach preconditioner output requested, but low-Mach preconditioner mode is not enabled."
```
Comment (benflexcompute) [inline tests/simulation/params/test_validators_params.py]: There are complaints from MC that the error message is not clear on how user should correct the settings. Let's try to have a more detailed message.

File: flow360/component/simulation/validation/validation_simulation_params.py [hunk 4]
```diff
@@ -39,6 +39,39 @@ def _check_consistency_wall_function_and_surface_output(v):
...
+
+    if not outputs:
+        return v
+
+    for output in outputs:
+        if not hasattr(output, "output_fields"):
+            continue
+        if "lowMachPreconditionerSensor" in output.output_fields.items:
+            raise ValueError(
+                "Low-Mach preconditioner output requested, but low-Mach preconditioner is not enabled."
```
Comment (benflexcompute) [inline flow360/component/simulation/validation/validation_simulation_params.py:69]:
```suggestion
                "Low-Mach preconditioner output requested, but models -> Fluid -> Navier Stokes solver -> Low Mach preconditioner is not enabled."
```

Comment (andrzej-krupka) [reply flow360/component/simulation/validation/validation_simulation_params.py]: Changed it to:

```
"Low-Mach preconditioner output requested, but low_mach_preconditioner is not enabled. "
"You can enable it via model.navier_stokes_solver.low_mach_preconditioner = True for a Fluid "
"model in the models field of the simulation object."
```

File: tests/simulation/params/test_validators_params.py (hunk 3 above)
Comment (andrzej-krupka) [reply tests/simulation/params/test_validators_params.py]: See next comment

File: flow360/component/simulation/validation/validation_simulation_params.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (andrzej-krupka) [reply flow360/component/simulation/validation/validation_simulation_params.py:41]: You're right. We had the same issue in other validators (`_check_consistency_wall_function_and_surface_output`, `_check_numerical_dissipation_factor_output`) as well so I fixed it there too...

**Review by maciej-flexcompute — APPROVED** [review]

**Pull request merged**